spark/v3.0/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DynamicFileFilterExec.scala

```diff
@@ -41,6 +43,11 @@ abstract class DynamicFileFilterExecBase(
   @transient
   override lazy val references: AttributeSet = AttributeSet(fileFilterExec.output)
 
+  override lazy val metrics = Map(
+    "totalFiles" -> SQLMetrics.createMetric(sparkContext, "total number of files"),
+    "hitFiles" -> SQLMetrics.createMetric(sparkContext, "number of files hit"),
+    "fileHitRate" -> SQLMetrics.createMetric(sparkContext, "file hit rate %"))
```

Comment:
I'm not sure that a derived metric is very valuable. I'd probably remove this and just keep the candidate and matching data files.

Reply:
done